lib/ansible/modules/cloud/azure/azure_rm_securitygroup.py

```diff
@@ -446,11 +446,11 @@ def create_rule_instance(rule):
     :return: SecurityRule
     '''
     return SecurityRule(
-        rule['protocol'],
-        rule['source_address_prefix'],
-        rule['destination_address_prefix'],
-        rule['access'],
-        rule['direction'],
+        protocol=rule.get('protocol', None),
```

Comment:
Note that you're doing two things at the same time:
1- Switching from positional to keyword syntax
2- Switching from `[]` syntax to `get` syntax

Only the first one is necessary for the bug you're trying to fix. Changing `[]` to `get` might change some `KeyError` exception to no exception and passing `None`, which might create later a `CloudError` instead. This is a change a behavior that is not required to fix the issue the title of this PR is.